apis/v1alpha1/httproute_types.go

```diff
@@ -328,79 +328,94 @@ type HTTPRouteMatch struct {
+	// Implementers are encouraged to define custom implementation types to
+	// extend the core API with implementation-specific behavior.
 	//
 	// +unionDiscriminator
-	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=100
-	Type string `json:"type"`
+	Type HTTPRouteFilterType `json:"type,omitempty"`
 
-	// ExtensionRef is an optional, implementation-specific extension to the
-	// "filter" behavior.  The resource may be "configmap" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myroutefilters" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".
-	// ExtensionRef MUST NOT be used for core and extended filters.
+	// RequestHeaderModifier defines a schema for a filter that modifies request
+	// headers.
+	//
+	// Support: Core
+	//
 	// +optional
-	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
-
-	// Filter-specific configuration definitions for core and extended filters
+	RequestHeaderModifier *HTTPRequestHeaderFilter `json:"requestHeaderModifier,omitempty"`
 
+	// RequestMirror defines a schema for a filter that mirrors requests.
+	//
+	// Support: Extended
+	//
 	// +optional
-	RequestHeader *HTTPRequestHeaderFilter `json:"requestHeader,omitempty"`
+	RequestMirror *HTTPRequestMirrorFilter `json:"requestMirror,omitempty"`
 
+	// ExtensionRef is an optional, implementation-specific extension to the
+	// "filter" behavior.  The resource may be "configmap" or an
+	// implementation-defined resource (for example, resource "myroutefilters"
+	// in group "networking.acme.io"). ExtensionRef MUST NOT be used for core
+	// and extended filters.
+	//
+	// Support: Implementation-specific
+	//
 	// +optional
-	RequestMirror *HTTPRequestMirrorFilter `json:"requestMirror,omitempty"`
+	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
 }
 
-// HTTPRequestHeaderFilter defines configuration for the
-// RequestHeader filter.
+// HTTPRouteFilterType identifies a type of HTTPRoute filter.
+// +kubebuilder:validation:Enum=RequestHeaderModifier;RequestMirror;Custom
+type HTTPRouteFilterType string
+
+const (
+	// RequestHeaderModifierHTTPRouteFilter can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	//
+	// Support: Core
+	RequestHeaderModifierHTTPRouteFilter HTTPRouteFilterType = "RequestHeaderModifier"
+
+	// RequestMirrorHTTPRouteFilter can be used to mirror HTTP requests to a
+	// different backend. The responses from this backend MUST be ignored by
+	// the Gateway.
+	//
+	// Support: Extended
+	RequestMirrorHTTPRouteFilter HTTPRouteFilterType = "RequestMirror"
+
+	// CustomHTTPRouteFilter should be used for configuring custom HTTP filters.
+	//
+	// Support: Implementation-specific
+	CustomHTTPRouteFilter HTTPRouteFilterType = "Custom"
```

Comment:
We usually resort to `ImplementationSpecific` literal string in most places in the API, any reason for using `Custom` here?

I like the conciseness of `Custom` over `ImplementationSpecific` but we should also be consistent. Should we consider changing to `Custom` in all places?

Reply:
This also relates to https://github.com/kubernetes-sigs/service-apis/issues/413

Reply:
I'll change to `ImplementationSpecific`. I went with `Custom` for brevity, but we can readdress across the API types in a separate PR.